docs/structured-streaming-programming-guide.md

```diff
@@ -2812,7 +2812,18 @@ See [Input Sources](#input-sources) and [Output Sinks](#output-sinks) sections f
 
 # Additional Information
 
-**Further Reading**
+## Configuration Options For Structured Streaming
+
+This section is for configurations which are only available for structured streaming, or they behave differently with batch query.
+
+- spark.sql.shuffle.partitions
```

Comment:
We do have it in `sql-programming-guide.md`. Shall we add some info there for now?

Reply:
IMHO, if something goes wrong with structured streaming, end users would try to review structured streaming guide doc, rather than sql programming guide doc. Could we wait for hearing more voices on this?

Reply:
What I am worried is about adding a new section, which is quite unusual. Usually we go for it when multiple instances are detected later.

Are there more instance to describe here specifically for structured streaming?

Reply:
I can revert adding a new section if you meant adding `##` on it. While gotcha looks more like funny, I will change it to `**Notes**`.

The rationalization on adding this to doc is, this restriction had been making lots of wondering around SO and user mailing list, as well as even a patch for fixing this. So all of end users who use structured streaming would be nice to see it at least once, even they skim the doc, so that they can remember and revisit the doc once they get stuck on this.